src/test/java/org/broadinstitute/consent/http/resources/InstitutionResourceTest.java

```diff
@@ -0,0 +1,160 @@
+package org.broadinstitute.consent.http.resources;
+
+import com.google.gson.Gson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.UserRole;
+import org.broadinstitute.consent.http.models.AuthUser;
+import org.broadinstitute.consent.http.models.Institution;
+import org.broadinstitute.consent.http.service.UserService;
+import org.broadinstitute.consent.http.service.InstitutionService;
+
+import org.broadinstitute.consent.http.util.InstitutionUtil;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import org.junit.Before;
+import org.junit.Test;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+public class InstitutionResourceTest {
+  private final AuthUser authUser = new AuthUser("test@test.com");
+  private final List<UserRole> adminRoles = Collections.singletonList(new UserRole(UserRoles.ADMIN.getRoleId(), UserRoles.ADMIN.getRoleName()));
+  private final List<UserRole> researcherRoles = Collections.singletonList(new UserRole(UserRoles.RESEARCHER.getRoleId(), UserRoles.RESEARCHER.getRoleName()));
+  private final User adminUser = new User(1, authUser.getName(), "Display Name", new Date(), adminRoles, authUser.getName());
+  private final User researcherUser = new User(1, authUser.getName(), "Display Name", new Date(), researcherRoles, authUser.getName());
+
+  @Mock InstitutionService institutionService;
+  @Mock UserService userService;
+  @Mock private UriInfo info;
+  @Mock private UriBuilder builder;
+  @Mock private User mockUser;
+  @Spy InstitutionUtil institutionUtil = new InstitutionUtil();
+
+  private InstitutionResource resource;
+
+  private Institution institutionSetup() {
+    Institution mockInstitution = initInsitutionModel();
+    mockInstitution.setCreateDate(new Date());
+    mockInstitution.setCreateUser(1);
+    mockInstitution.setUpdateDate(new Date());
+    mockInstitution.setUpdateUser(1);
+    mockInstitution.setId(1);
+    return mockInstitution;
+  }
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    resource = new InstitutionResource(userService, institutionService, institutionUtil);
+  }
+
+  private Institution initInsitutionModel() {
+    Institution mockInstitution = new Institution();
+    mockInstitution.setName("Test Name");
+    return mockInstitution;
+  }
+
+  private void initResource() {
+    resource = new InstitutionResource(userService, institutionService, institutionUtil);
```

Comment:
Having this in two places (here and `setUp`) is a bit confusing. As it stands now, resource will be initialized twice for any method that calls `initResource`. It would be cleaner to only have it here, and call `initResource` for every test method.